functional_tests/scylla_operator/test_functional.py

```diff
@@ -272,3 +274,57 @@ def test_orphaned_services_multi_rack(db_cluster):
 
     db_cluster.wait_for_pods_readiness(pods_to_wait=2, total_pods=len(db_cluster.nodes))
     assert not get_orphaned_services(db_cluster), "Orphaned services were found after decommission"
+
+
+def test_ha_update_spec_while_rollout_restart(db_cluster: ScyllaPodCluster):
+    """
+    Cover the issue https://github.com/scylladb/scylla-operator/issues/410
+    Validate that cluster resources can be updated while the operator StatefulSet is rolling out.
+    - update cluster specification a few time
+    - start rollout restart in parallel with the update
+    - validate that the cluster specification has been updated
+    """
+    to_terminate = threading.Event()
+    value = 1048576
+    errors = []
+
+    def change_cluster_spec():
+        nonlocal value
+        nonlocal errors
+        while not to_terminate.wait(0.1):
+            try:
+                db_cluster.replace_scylla_cluster_value('/spec/sysctls', [f"fs.aio-max-nr={value + 1}"])
+                # increase the value just when the sysctls spec value has been updated - to prevent the situation when
+                # value was increased, but sysctls spec value was not updated
+                value += 1
+            except Exception as error:  # pylint: disable=broad-except
+                errors.append(str(error))
+
+    change_cluster_spec_thread = threading.Thread(target=change_cluster_spec, daemon=True)
+    log.info("Start update cluster specification")
+    change_cluster_spec_thread.start()
+
+    log.info("Start rollout restart")
+    scylla_operator_rollout_restart(db_cluster)
+    status = wait_for_scylla_operator_rollout_complete(db_cluster)
+    assert not status, "Rollout restart failed. Reasons: {}".format('\n'.join(status))
```

Comment:
Lines 309-310 expect that "status" is empty when everything is ok. It sounds very strange to me. Not intuitive.
If we keep behavior maybe we should rename this var? Instead of "status" to something like "operator_rollout_errors"?
Also, rename above "errors" to something like "crd_update_errors"...

Reply:
changed